src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTest.php

```diff
@@ -37,10 +41,99 @@ public function testForward()
         $container->expects($this->at(0))->method('get')->will($this->returnValue($requestStack));
         $container->expects($this->at(1))->method('get')->will($this->returnValue($kernel));
 
-        $controller = new Controller();
+        $controller = new TestController();
```

Comment:
I haven't used reflection so far, so I am not very familiar with it. @fabpot Your suggestion sounds like it would avoid the need of the `TestController` class completely. If I am not wrong, it would only avoid the overriding of the methods. Maybe that was your point. In this case it would not simplify the testcode IMO, since I have to change 

```php
$this->assertSame($controller->getUser(), $user);
```

to

```php
 $m = new \ReflectionMethod('Symfony\Bundle\FrameworkBundle\Controller\Controller', 'getUser');
 $m->setAccessible(true);
 $this->assertSame($m->invoke($controller), $user);
```
        
where I still need

```php
$controller = new TestController();
$controller->setContainer($this->getContainerWithTokenStorage($token));
```
        
By avoiding adding all the tested methods of `Controller` progressively, the test methods itself would become unlike more verbose. Although it could be reduced with a helper method, for me, the current version is simpler (for readers, writers and runtime).

@xabbuh Using `__call` is indeed an option, but has the drawback that the IDE complains about calling the protected method then.

```php
class TestController extends Controller
{
    function __call($name, $arguments)
    {
        return call_user_func_array('parent::'.$name, $arguments);
    }
}
```

Let me know if I should change the PR though to either use reflection or the __call method.

To address @iltar's comment... What about capitalizing the abilities of traits in PHP 5.4 for SF 3.0 by providing traits with the shortcut methods, so developers can use the methods without extending a base Controller?